app/controllers/admin/reports_controller.rb

```diff
@@ -20,14 +20,17 @@ def index
     end
 
     def show
-      @report = report_class.new(spree_current_user, params, render: render_data?)
+      @report = report_class.new(spree_current_user, params, render: false)
       @rendering_options = rendering_options # also stores user preferences
 
-      if render_data?
-        render_in_background
-      else
-        show_report
-      end
+      show_report
+    end
+
+    def create
+      @report = report_class.new(spree_current_user, params, render: true)
+      @rendering_options = rendering_options # also stores user preferences
+
+      render_in_background
     end
```

Comment:
This isn't necessary but the old structure of combining the two actions in one method had bothered me for a while.

Reply:
Nice, the change looks surprisingly simple too.